packages/assistant/src/RPA/Assistant/flet_client.py

```diff
@@ -169,20 +169,36 @@ def on_session_created(conn, session_data):
             # pylint: disable=raise-missing-from
             raise TimeoutException("Reached timeout while waiting for Assistant Dialog")
 
-    def _make_flet_event_handler(self, name: str):
-        def change_listener(e: ControlEvent):
-            self.results[name] = e.data
-            e.page.update()
+    def _make_flet_event_handler(self, name: str, handler: Optional[Callable] = None):
+        """Add flet event handler to record the element's data whenever content changes
+        if ``handler`` is provided also call that.
+        """
+
+        # We don't want the if inside the change_listener so it doesn't have to run on
+        # every on_change event
+        if not handler:
+
+            def change_listener(e: ControlEvent):
+                self.results[name] = e.data
+
+        else:
+
+            def change_listener(e: ControlEvent):
+                handler(e)
+                self.results[name] = e.data
```

Comment:
```suggestion
            def change_listener(e: ControlEvent):
                if handler:
                    handler(e)
                self.results[name] = e.data
```

Reply:
If you do this one then also remove the comment about why I did the earlier solution 😅 .  (Lines 177-178:

```
# We don't want the if inside the change_listener so it doesn't have to run on
# every on_change event
```